### PR TITLE
Upgrading cassandra java driver from 4.6.0 to 4.11.0

### PR DESCRIPTION
**Pull request by Sfurti-yb**
Changes made and steps followed: [Google doc](https://docs.google.com/document/d/1CESAraQhZ0nqtDKEFnl48yVddHTDYmDzqkGPzAy-4zg/edit?usp=sharing)

### CODE REVIEW

File: core/src/main/java/com/yugabyte/oss/driver/internal/core/loadbalancing/PartitionAwarePolicy.java [hunk 1]
```diff
@@ -30,7 +30,7 @@
 import com.datastax.oss.driver.api.core.type.SetType;
 import com.datastax.oss.driver.api.core.type.UserDefinedType;
 import com.datastax.oss.driver.internal.core.loadbalancing.DefaultLoadBalancingPolicy;
-import com.datastax.oss.driver.internal.core.util.collection.QueryPlan;
+import com.datastax.oss.driver.internal.core.util.collection.QueryPlanNew;
```
Comment (nchandrappa) [inline core/src/main/java/com/yugabyte/oss/driver/internal/core/loadbalancing/PartitionAwarePolicy.java]: Why did we need a new class `QueryPlanNew`, could we have used `QueryPlanBase`. This is a major change, probably we'll need to add a test case if there is no integration test coverage.

Comment (nchandrappa) [reply core/src/main/java/com/yugabyte/oss/driver/internal/core/loadbalancing/PartitionAwarePolicy.java]: We need to verify what each of these new QueryPlan implementations provides - QueryPlanBase, LazyQueryPlan, EmptyQueryPlan.

File: core/src/main/resources/reference.conf [hunk 2]
```diff
@@ -156,7 +156,25 @@ datastax-java-driver {
...
+    #   datacenter from the provided contact points, if possible; if that fails, it will throw an
+    #   error during session initialization. This policy is intended mostly for ETL tools and
+    #   should not be used by normal applications.
+    # - `BasicLoadBalancingPolicy`: similar to `DefaultLoadBalancingPolicy`, but does not have
+    #   the slow replica avoidance mechanism. More importantly, it is the only policy capable of
+    #   operating without local datacenter defined, in which case it will consider nodes in the
+    #   cluster in a datacenter-agnostic way. Beware that this could cause spikes in
+    #   cross-datacenter traffic! This policy is provided mostly as a starting point for users
+    #   wishing to implement their own load balancing policy; it should not be used as is in normal
+    #   applications.
```
Comment (nchandrappa) [inline core/src/main/resources/reference.conf:177]: Add a comment explaining `PartitionAwarePolicy`similar to the other LoadBalancing policies in `reference.conf`.

**Review by nchandrappa — COMMENTED** [review]
Please see the review comments.

File: core/src/main/java/com/yugabyte/oss/driver/internal/core/loadbalancing/PartitionAwarePolicy.java (hunk 1 above)
Comment (Sfurti-yb) [reply core/src/main/java/com/yugabyte/oss/driver/internal/core/loadbalancing/PartitionAwarePolicy.java]: QueryPlanBase is also an abstract class and hence cannot be instantiated in the PartitionAwarePolicy.java. I will look into LazyQueryPlan and EmptyQueryPlan.

File: core/src/main/resources/reference.conf (hunk 2 above)
Comment (Sfurti-yb) [reply core/src/main/resources/reference.conf:177]: Will do.

**Comment by nchandrappa** [comment]
Updates for Review comments look good. We can merge and release `4.11.0-yb-x` driver.

**Pull request merged**